exchanges/bitmex/bitmex_test.go

```diff
@@ -679,3 +681,34 @@ func TestGetDepositAddress(t *testing.T) {
 		}
 	}
 }
+
+// TestWsAuth dials websocket, sends login request.
+func TestWsAuth(t *testing.T) {
+	b.SetDefaults()
+	TestSetup(t)
+	if !b.Websocket.IsEnabled() && !b.AuthenticatedAPISupport || !areTestAPIKeysSet() {
+		t.Skip(exchange.WebsocketNotEnabled)
+	}
+	var err error
+	var dialer websocket.Dialer
+	b.WebsocketConn, _, err = dialer.Dial(b.Websocket.GetWebsocketURL(),
+		http.Header{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	b.Websocket.DataHandler = make(chan interface{}, 999)
+	b.Websocket.TrafficAlert = make(chan struct{}, 999)
+	go b.wsHandleIncomingData()
+	defer b.WebsocketConn.Close()
+	err = b.websocketSendAuth()
+	if err != nil {
+		t.Error(err)
+	}
+	timer := time.NewTimer(3 * time.Second)
+	select {
+	case <-b.Websocket.DataHandler:
```

Comment:
Its to ensure there is a proper response and not a timeout since I'm expecting a response. I'll flesh it out to read the response and act if its not to my expectations (if applicable)